go/apps/bulk_message/definition.py

```diff
@@ -19,11 +24,29 @@ def check_disabled(self):
                 " messages attached to this conversation.")
 
     def perform_action(self, action_data):
-        return self.send_command(
-            'bulk_send', batch_id=self._conv.batch.key,
-            msg_options={}, content=action_data['message'],
-            delivery_class=action_data['delivery_class'],
-            dedupe=action_data['dedupe'])
+        if action_data['scheduled_datetime'] is None:
+            return self.send_command(
+                'bulk_send', batch_id=self._conv.batch.key,
+                msg_options={}, content=action_data['message'],
+                delivery_class=action_data['delivery_class'],
+                dedupe=action_data['dedupe'])
+        else:
+            task = Task.objects.create(
+                account_id=self._conv.user_api.user_account_key,
+                label='Bulk Message Send',
+                task_type=Task.TYPE_CONVERSATION_ACTION,
+                task_data={
+                    'action_name': 'bulk_send',
+                    'action_kwargs': {
+                        'batch_id': self._conv.batch.key,
+                        'msg_options': {},
+                        'content': action_data['message'],
+                        'delivery_class': action_data['delivery_class'],
+                        'dedupe': action_data['dedupe'],
+                    },
+                },
+                scheduled_for=action_data['scheduled_datetime'])
+            task.save()
```

Comment:
If we do move support for scheduling into the base action, we can add a `schedule_command` method that is similar to `send_command` and that actions will need to call?